packages/loader/container-loader/src/container.ts

```diff
@@ -635,8 +635,6 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
     }
 
     public setAutoReconnect(reconnect: boolean) {
-        assert(this.resumedOpProcessingAfterLoad);
```

Comment:
I'm 99% sure that's not enough.
Let's look at couple cases:
1. non-attached  container should not call this.connectToDeltaStream below - this will explode I think
2. If resume was not called for attached container, I think we should not call this.connectToDeltaStream(). I.e. this API only sets the policy. If combination of policy & current state (attached and resumed client) results in state machinery to start reconnect, then we should call this.connectToDeltaStream

Reply:
1.) Yes good catch. connectToDeltaStream should not be called on unattached container.
2.) attach calls resume internally.So if we are attached then we must be resumed. So in that case if reconnect is true, then we should ca;; connectToDeltaStream to be true.

Reply:
Also, now I have moved the assert to if condition. Because attached container must have been resumed.